### ISSUE TITLE
Husky 6 - Docs - Update docs to inform users on how to set hook scripts as executable with Git

### ISSUE THREAD

**Issue by ThomasOrtiz** [typicode/husky#934]
**Context**
The V6 install guide worked well for Windows users but Unix based systems needed the hook scripts to be explicitly marked as executable at the Git level even with `"prepare": "husky install"`.

If hook scripts aren't marked as executable Unix users got the following git warning:
![image](https://user-images.githubusercontent.com/10877317/113583524-be791880-95f7-11eb-8355-0ff84524c977.png)

**Proposal**
- Update docs in at least the [Hooks not running](https://typicode.github.io/husky/#/?id=hooks-not-running) section and maybe even in the main install guide to mark hooks scripts as executable.
- ❓ should Husky mark scripts it generates as git executable?

Webpack example
![image](https://user-images.githubusercontent.com/10877317/113583288-778b2300-95f7-11eb-9552-062042c48e46.png)

**Why**
Because V4 to V6 migration requires a lot more effort on consumers part the process should be as smooth as possible.

**Executable files and Git**
By default, all files in Git are marked as non executable.
Thats generally fine BUT for code-quality scripts (say `husky git hooks`) we want to ensure that they're executable when the repo is cloned.

Luckily Git supports marking files as executable!

```jsonc
// Register an untracked file to Git's index and say that its executable
git add --chmod=+x pre-commit

// Set an existing file to be executable
git update-index --chmod=+x pre-commit

// Set an existing file to NOT be executable
git update-index --chmod=-x pre-commit
```

You can also view a tracked file's Git permissions with the following command
```jsonc
git ls-files --stage

// Example output in the .husky directory
// 100644 for regular files
// 100755 for executable ones
100644 31354ec1389994b5f6708c7d915fdcc6bb76ba6e 0       .gitignore
100755 d37daa075e22b05c2addeeb8b414387d05ca5e84 0       pre-commit
```

**Comment by typicode** [comment]
Thanks for the detailed feedback :+1: 

> should Husky mark scripts it generates as git executable?

That's already the case when you use `husky add` or `husky set`. So no further action should be needed.

> Update docs in at least the Hooks not running section and maybe even in the main install guide to mark hooks scripts as executable.

Good idea for the not running section.

**Comment by ThomasOrtiz** [comment]
> `That's already the case when you use husky add or husky set. So no further action should be needed.`

I don't believe that Husky's Add/Set fully set the permissions. This [code (below snippet)](https://github.com/typicode/husky/blob/main/packages/husky/src/index.ts#L70) does indeed set the file system permissions, however, when you commit the hook file to your repo the file doesn't seem to be marked with Git to be executable.

```sh
fs.writeFileSync(file, `#!/bin/sh. "$(dirname "$0")/_/husky.sh" ${cmd}`, { mode: 0o0755 })
```

@typicode

**Comment by ThomasOrtiz** [comment]
For example:
1. Create a new repo
2. Run `npx husky-init && npm install`
3. Commit changes
4. Notice that the pushed hook file is not marked as executable.

As such anyone that clones the repo will have to manually set the file as executable whereas if the file was added to git like `git add --chmod=+x pre-commit` or `git update-index --chmod=+x pre-commit` then on clone the file will be marked as executable.

![image](https://user-images.githubusercontent.com/10877317/113762424-8dc3dc80-96e6-11eb-9e0c-f3575590277c.png)

**Comment by ricmello** [comment]
@ThomasOrtiz I'm not sure if marking files as executable is the best option. My concern is that any hook created manually will need to be set as executable in git, which can be tricky for some users.

My suggestion is change the `husky install` script to set the hooks as executable when it runs so the users will not need to care about it, and any `npm install` can solve possible permission errors.

What do you think?

**Comment by ThomasOrtiz** [comment]
@ricmello the more that can be automated the better so having it part of `husky install` would be ideal.

**Comment by ThomasOrtiz** [comment]
@typicode Thoughts on this after the further discussion?

**Comment by mordv** [comment]
Any progress on this issue? It was very frustrating starting a new project with husky v6 and then facing the problem with file permissions and fixing it locally on each developer's machine.

husky v4 _just_ worked.
